sql/analyzer/replace_sort.go

```diff
@@ -291,6 +297,21 @@ func isValidSortFieldOrder(sfs sql.SortFields) bool {
 	return true
 }
 
+// isNonOverlapping checks if the ranges in a RangeCollection that are part of the sortfield exprs are non-overlapping
+// This function assumes that the sort field exprs are a valid prefix of the index columns
+func isNonOverlapping(sfExprs []sql.Expression, ranges sql.RangeCollection) bool {
```

Comment:
zach usually tells me to avoid not checks, easier to follow logic asserting the correctness of certain conditions